### PR TITLE
Added support for beta vs ga releases

### PR DESCRIPTION
**Pull request by sthirugn**
Testing is blocked and is reported

### CODE REVIEW

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -1178,6 +1180,9 @@ def product_install(distribution, create_vm=False, certificate_url=None,
     # Fetch the Satellite Version information.
     satellite_version = os.environ.get('SATELLITE_VERSION')
 
+    # Fetch the Satellite Release information.
+    satellite_release = os.environ.get('SATELLITE_RELEASE')
```
Comment (elyezer) [inline automation_tools/__init__.py]: if the release is not available what should we do?

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -1261,6 +1266,7 @@ def product_install(distribution, create_vm=False, certificate_url=None,
         host=host,
         sam=distribution.startswith('sam'),
         sat_version=satellite_version,
+        sat_release=satellite_release,
```
Comment (elyezer) [inline automation_tools/__init__.py]: Is this prepared to receive `None` as release? Otherwise we should make sure a proper release is passed.

Comment (elyezer) [inline automation_tools/__init__.py]: I commenting before reading the remaining of the code. Never mind about this.

**Comment by elyezer** [comment]
I think the solution for this would be simpler. Since the package installed is the same what about checking if the satellite-installer or katellite-installer or foreman-installer is available? This can be handled on the installer task without the need of adding more required arguments and environment variables.

What do you think about this?

**Comment by sthirugn** [comment]
I spoke to @elyezer about his recommendation of checking the presence of satellite-installer/katello-installer.  But at this time, it is good to have explicit checks since there are lot of changes happening between beta and GA.  We can merge this PR and plan any improvements for later.  @elyezer also agreed to this in irc.

**Comment by sthirugn** [comment]
ready for final review/merge

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -1863,7 +1869,16 @@ def katello_installer(debug=False, sam=False, verbose=True, sat_version='6.2',
         sat_version = ''
     if sat_version == '6.2':
         proxy = 'foreman-proxy'
-        installer = 'foreman'
+        if sat_release == 'GA':
```
Comment (elyezer) [inline automation_tools/__init__.py]: what about ignoring the case here? `sat_release.lower() == 'ga'`

**Comment by elyezer** [comment]
ACK pending comment.

File: automation_tools/__init__.py [hunk 4]
```diff
@@ -955,6 +955,8 @@ def downstream_install(admin_password=None, run_katello_installer=True):
         Optional, defaults to 'changeme'. Foreman admin password.
     BASE_URL
         URL for the compose repository.
+    SATELLITE_VERSION
+        Satellite version.
```
Comment (sthirugn) [inline automation_tools/__init__.py]: This is just docstring update - this is not introduced by this PR.

**Comment by sthirugn** [comment]
Comments updated and rebased

**Comment by elyezer** [comment]
ACK

**Pull request merged**